src/FA/FiniteAutomata.java

```diff
@@ -0,0 +1,137 @@
+package FA;
+
+import domain.Pair;
+
+import java.io.*;
+import java.util.*;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.stream.Collectors;
+
+public class FiniteAutomata {
+    private List<String> Q;
+    private List<String> E;
+    private HashMap<Pair<String, String>, List<String>> D;
+    private String q0;
+    private List<String> F;
```

Comment:
Except for the fact that i know what does represent because of the nature of the problem, nobody will when reading the code